src/main/java/lotto/controller/GameController.java

```diff
@@ -0,0 +1,104 @@
+    private final OutputView outputView;
+    private final GameService gameService;
+    private final LottoService lottoService;
+
+    public GameController(InputView inputView, OutputView outputView, GameService gameService, LottoService lottoService) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.gameService = gameService;
+        this.lottoService = lottoService;
+    }
+
+    public void run() {
+        purchaseLottos();
+        issueLottos();
+        inputWinningLotto();
+        inputBonusNumber();
+        outputWinningStatistics();
+        outputProfitRate();
+    }
+
+    public void purchaseLottos() {
+        String input;
+        do {
+            outputView.requestPurchaseAmount();
+            input = inputView.scanPurchaseAmount();
+        } while (isInvalidPurchaseAmount(input));
+    }
+
+    public void issueLottos() {
+        try {
+            lottoService.initLottos(gameService.createLottos());
+        } catch (IllegalArgumentException e) {
+            outputView.printException(e.getMessage());
+            throw e;
+        }
+        outputView.printPurchaseNumber(gameService.getPurchaseNumber());
+        outputView.printLottos(lottoService.getLottosOutput());
+    }
+
+    public void inputWinningLotto() {
+        lottoService.initWinningNumbers();
+        String input;
+        do {
+            outputView.requestWinningLotto();
+            input = inputView.scanWinningLotto();
+        } while (isInvalidWinningLotto(input));
+    }
+
+    public void inputBonusNumber() {
+        String input;
+        do {
+            outputView.requestBonusNumber();
+            input = inputView.scanBonusNumber();
+        } while (isInvalidBonusNumber(input));
+    }
+
+    public void outputWinningStatistics() {
+        gameService.initRankStatistics(lottoService.createStatistics());
+        outputView.printRankStatistics(gameService.getRankStatisticsOutput());
+    }
+
+    public void outputProfitRate() {
+        outputView.printProfitRate(gameService.getProfitRate());
+    }
+
+    private boolean isInvalidPurchaseAmount(String input) {
+        try {
+            gameService.initPurchaseAmount(input);
+        } catch (IllegalArgumentException e) {
+            outputView.printException(e.getMessage());
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isInvalidWinningLotto(String input) {
+        try {
+            lottoService.initWinningLotto(input);
+        } catch (IllegalArgumentException e) {
+            outputView.printException(e.getMessage());
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isInvalidBonusNumber(String input) {
+        try {
+            lottoService.initBonusNumber(input);
+        } catch (IllegalArgumentException e) {
+            outputView.printException(e.getMessage());
+            return true;
+        }
+        return false;
+    }
```

Comment:
위에 제가 말씀드린대로 do - while 을 고친다면 모두 사라질 수 있는 메서드이겠군요. 그런데 만약 이 메서드들이 필요했다면 같은 구조를 가지고 있다는 점에서 함수형 포인터를 활용한 리펙토링을 고려했으면 더 중복없는 재사용 코드가 되었을 것 같습니다!

```suggestion
   private boolean isInvalidInput(String input, Function<String, Void> serviceMethod) {
        try {
            serviceMethod.apply(input);
        } catch (IllegalArgumentException e) {
            outputView.printException(e.getMessage());
            return true;
        }
        return false;
    }

    private boolean isInvalidPurchaseAmount(String input) {
        return isInvalidInput(input, gameService::initPurchaseAmount);
    }

    private boolean isInvalidWinningLotto(String input) {
        return isInvalidInput(input, lottoService::initWinningLotto);
    }

    private boolean isInvalidBonusNumber(String input) {
        return isInvalidInput(input, lottoService::initBonusNumber);
    }
```

Reply:
함수형 인터페이스를 사용해야겠다고 생각해본적이 없었던 것 같습니다. 그런데 이렇게도 활용할수가 있군요!! 정말 가독성 뿐만 아니라 코드 재사용성이 좋아진 것이 실감이 되네요. 유익한 정보 공유해주셔서 감사합니다 :)